src/components/RunInCloudDialog/states/index.tsx

```diff
@@ -0,0 +1,32 @@
+import { exhaustive } from '@/utils/typescript'
+import { SignIn } from './SignIn'
+import { RunInCloudState } from './types'
+import { Error } from './Error'
+import { LoadingMessage } from '@/components/Profile/LoadingMessage'
+
+interface RunInCloudStatesProps {
+  state: RunInCloudState
+  onAbort: () => void
+}
+
+export function RunInCloudStates({ state }: RunInCloudStatesProps) {
+  switch (state.type) {
+    case 'initializing':
+      return <LoadingMessage>Initializing...</LoadingMessage>
+
+    case 'sign-in':
+      return <SignIn />
+
+    case 'preparing':
+    case 'uploading':
+    case 'starting':
+    case 'started':
+      return <LoadingMessage>Uploading...</LoadingMessage>
```

Comment:
Should we add more visual niceties to the window ?
We could consider adding a grot and/or showing some info on where we are uploading ? (project id, etc..) 🤔

Reply:
Maybe this one?

![image](https://github.com/user-attachments/assets/080f70e5-076c-402c-ba19-488f9cfff83a)

Reply:
I added the grot to the dialog. It's a bit more fun now.

Tried including the project name but the text just becomes too wide. Besides, since we open the run they can already see which project they're in. So I would prefer to merge as-is and maybe revisit this when we discuss having configurable projects.